code/modules/flufftext/Hallucination.dm

```diff
@@ -325,151 +305,195 @@ GLOBAL_LIST_INIT(hallucinations_major, list(
 	QDEL_NULL(bubblegum)
 	return ..()
 
-/datum/hallucination/singularity_scare
-	//Singularity moving towards you.
-	//todo Hide where it moved with fake space images
-	var/obj/effect/hallucination/simple/singularity/s = null
-	cost = 75
-
-/datum/hallucination/singularity_scare/New(mob/living/carbon/T, forced = TRUE)
-	set waitfor = FALSE
-	..()
-	var/turf/start = get_turf(T)
-	var/screen_border = pick(SOUTH,EAST,WEST,NORTH)
-	for(var/i in 1 to 13)
-		start = get_step(start,screen_border)
-	feedback_details += "Source: [start.x],[start.y],[start.z]"
-	s = new(start,target)
-	s.parent = src
-	for(var/i in 1 to 13)
-		sleep(10)
-		s.forceMove(get_step(get_turf(s),get_dir(s,target)))
-		s.Show()
-		s.Eat()
-	qdel(s)
-
-/obj/effect/hallucination/simple/singularity
-	image_icon = 'icons/effects/224x224.dmi'
-	image_state = "singularity_s7"
-	image_layer = MASSIVE_OBJ_LAYER
-	px = -96
-	py = -96
-	var/datum/hallucination/singularity_scare/parent
-
-/obj/effect/hallucination/simple/singularity/proc/Eat(atom/OldLoc, Dir)
-	var/target_dist = get_dist(src,target)
-	if(target_dist<=3) //"Eaten"
-		target.set_screwyhud(SCREWYHUD_DEAD)
-		target.SetUnconscious(160)
-		addtimer(CALLBACK(parent, /datum/hallucination/.proc/wake_and_restore), rand(30, 50))
-
 /datum/hallucination/battle
-	cost = 15
 
-/datum/hallucination/battle/New(mob/living/carbon/T, forced = TRUE, battle_type)
+/datum/hallucination/battle/New(mob/living/carbon/C, forced = TRUE, battle_type)
 	set waitfor = FALSE
 	..()
-	var/hits = rand(3,6)
+	var/list/turf/sourcelocs = list()
+	for(var/turf/open/T in range(world.view+1,target)-range(world.view-2,target))
```

Comment:
If you're just picking one random offset is better than iterating all these (twice). Could probably pick from off-view locations only alternatively.